bdk_rpc_wallet_example/src/rpc.rs

```diff
@@ -0,0 +1,314 @@
+pub struct RpcConfig {
+    url: String,
+    auth: Auth,
+    network: Network,
+}
+
+impl RpcConfig {
+    pub fn new(url: String, auth: Auth, network: Network) -> Self {
+        Self { url, auth, network }
+    }
+}
+
+/// Return list of bitcoin core wallet directories
+fn list_wallet_dirs(client: &Client) -> Result<Vec<String>, RpcError> {
+    #[derive(serde::Deserialize)]
+    struct Name {
+        name: String,
+    }
+    #[derive(serde::Deserialize)]
+    struct CallResult {
+        wallets: Vec<Name>,
+    }
+
+    let result: CallResult = client.call("listwalletdir", &[])?;
+    Ok(result.wallets.into_iter().map(|n| n.name).collect())
+}
+
+/// Create unique wallet name from [`KeychainTxOutIndex`].
+pub fn derive_name_from_keychainindex<K: Debug + Clone + Ord>(
+    txout_index: &KeychainTxOutIndex<K>,
+) -> String {
+    let data = txout_index
+        .keychains()
+        .iter()
+        .fold(Vec::new(), |mut accum, item| {
+            let mut bytes = item.1.to_string().as_bytes().to_vec();
+            accum.append(&mut bytes);
+            return accum;
+        });
+    sha256Hash::hash(&data[..]).to_hex()[0..10].to_string()
+}
+
+/// Import script_pubkeys into bitcoin core wallet.
+pub fn import_multi<'a>(
+    client: &Client,
+    scripts: impl Iterator<Item = &'a Script>,
+) -> Result<(), RpcError> {
+    let requests = scripts
+        .map(|script| ImportMultiRequest {
+            timestamp: ImportMultiRescanSince::Now,
+            script_pubkey: Some(ImportMultiRequestScriptPubkey::Script(script)),
+            watchonly: Some(true),
+            ..Default::default()
+        })
+        .collect::<Vec<_>>();
+
+    let options = ImportMultiOptions {
+        rescan: Some(false),
+    };
+    for import_multi_result in client.import_multi(&requests, Some(&options))? {
+        if let Some(err) = import_multi_result.error {
+            return Err(err.into());
+        }
+    }
+    Ok(())
+}
+
+/// A Bitcoin Core RPC Client struct that can be used to sync the [`KeychainTxOutIndex`]
+pub struct RpcClient {
+    client: Client,
+}
+
+impl Deref for RpcClient {
+    type Target = Client;
+    fn deref(&self) -> &Self::Target {
+        &self.client
+    }
+}
+
+impl RpcClient {
+    /// Initialize [`RpcClient`] from a [`RpcConfig`] and [`KeychainTxOutIndex`].
+    /// This creates a wallet inside bitcoin core, with a deterministic name
+    /// derived from the hash of descriptors within the [`KeychainTxOutIndex`].
+    /// Then imports all the spks from the index into the newly created core wallet.
+    pub fn init_for_tracker<K: Debug + Clone + Ord>(
+        config: &RpcConfig,
+        txout_index: &KeychainTxOutIndex<K>,
+    ) -> Result<Self, RpcError> {
+        let wallet_name = derive_name_from_keychainindex(txout_index);
+        let wallet_url = format!("{}/wallet/{}", config.url, wallet_name);
+        let client = Client::new(wallet_url.as_str(), config.auth.clone().into())?;
```

Comment:
Please make the caller create the client and then call methods on it. That way we don't need to have any config defined in this library.